src/main/java/com/sk89q/worldguard/sponge/listener/WorldGuardBlockListener.java

```diff
@@ -83,13 +87,15 @@ public void onBlockBreak(BreakBlockEvent event) {
      * Called when fluids flow.
      */
     @Listener
-    public void onBlockFromTo(final PlaceBlockEvent event) {
-        Location<World> block = event.getCause().getFirst(Location.class).orNull();
-        if (block == null) return;
+    public void onBlockFromTo(final ChangeBlockEvent.Place event) {
+        Location<World> block = event.getCause().first(Location.class).orElse(null);
```

Comment:
Location will never be in the cause, this should be looping through each Transaction<BlockSnapshot>